### PR TITLE
Move system tests to `tests`

### PR DESCRIPTION
**Pull request by kachkaev**
## 🌟 What is the purpose of this PR?

This PR extracts backend integration tests and Playwright tests into a new top-level folder called `tests`. Yarn workspaces are renamed accordingly, i.e. they now use `@tests/*` namespace.

Note that the top-level `tests` folder is for system tests (integration, E2E). Unit tests are located within corresponding workspaces, next to the files that they cover (`foo.ts` → `foo.test.ts`).

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203543021352032/1203324923411371/f) _(internal)_

## 🐾 Next steps

Move testable workspaces (api, frontend, etc.) into their new locations

## 🛡 What tests cover this?

- CI

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -280,6 +280,6 @@ jobs:
         name: Upload coverage to https://app.codecov.io/gh/hashintel/hash
         with:
           fail_ci_if_error: true
-          files: packages/hash/api/coverage/lcov.info,packages/hash/backend-utils/coverage/lcov.info,packages/hash/frontend/coverage/lcov.infopackages/hash/shared/coverage/lcov.info,
+          files: packages/hash/api/coverage/lcov.info,packages/hash/backend-utils/coverage/lcov.info,packages/hash/frontend/coverage/lcov.info,packages/hash/shared/coverage/lcov.info,
```
Comment (kachkaev) [inline .github/workflows/ci.yml:283]: This typo fix is unrelated, just chucking it so that we don't forget.

**Review by CiaranMn — APPROVED** [review]
